src/bootstrap/builder.rs

```diff
@@ -1111,6 +1111,15 @@ impl<'a> Builder<'a> {
             );
         }
 
+        // If Control Flow Guard is enabled, pass the `control_flow_guard=checks` flag to rustc.
+        // This is not needed for stage 0.
+        if (mode == Mode::Std || mode == Mode::Codegen)
```

Comment:
Can we have some commentary here as to why we're not building the compiler or tools with this flag?

Reply:
Thanks for the question! Just to check my thinking: the main aim here is to enable CFG checks for outputs that could linked into the final binaries generated by rustc. Is this the correct way to achieve that?

My assumption is that the compiler and tools themselves wouldn't really be targets for attack, but if this is not the case, we could certainly enable CFG here too, probably as a separate flag (although I haven't yet tested this).

Does this make sense? If so, I'll update the comment.

Reply:
Replied to the comment above - but response applies to both I think.